### ISSUE TITLE
Make preview seekbar only generate the preview once

### ISSUE THREAD

**Issue by Luna712** [recloudstream/cloudstream#657]
### Describe your suggested feature

I think it would be more efficient if it only generated once, so if the source is changed it is maintained without having to regenerate again. It could maybe check the length of the video and if it (and the title) matches exactly then maintain the same preview, at least in the context of the same player. (maybe not if you back out to a different source entirely)

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by asl97** [comment]
If you want it to be efficient, it would be best if the preview can be extracted from the source or provided by the provider,

but that would require quite a bit of sources specific code for very little benefits.

**Comment by Luna712** [comment]
> I already thought about it, but thought it was best to make it regenerate when switching source even on the same episode would be best in case of bad/different sources

Yeah, thats why I mentioned maybe check the exact length and if it matches exactly there is a 99% probability it is the exact same video, so thought that would be a good way, but I understand. Thats fine. Thanks!
